src/aiida_pseudo/cli/params/types.py

```diff
@@ -54,13 +55,34 @@ def complete(self, _, incomplete):
 
 
 class PseudoPotentialFamilyParam(GroupParamType):
-    """Parameter type for `click` commands to define an instance of a `PseudoPotentialFamily`."""
+    """Parameter type for ``click`` commands to define an instance of a ``PseudoPotentialFamily``."""
 
     name = 'pseudo_family'
 
+    def __init__(self, exclude: typing.Optional[typing.List[str]] = None, **kwargs):
```

Comment:
```suggestion
    def __init__(self, exclude: list[str] | None = None, **kwargs):
```
You can use this notation if you put `from __future__ import annotations` at the top of the file (just below the docstring). We should start to use this everywhere, it is so much more readable.

Reply:
Would 
```python
def convert(self, value, param, ctx) -> typing.Union[pathlib.Path, bytes]:
```

Then also become
```python
def convert(self, value, param, ctx) -> pathlib.Path | bytes:
```

Do you have any good tutorial / reference on this?

Reply:
yes, the `|` operator functions the same as `typing.Union`. Best reference are official docs: https://docs.python.org/3/library/typing.html